types/handsontable/index.d.ts

```diff
@@ -401,8 +403,100 @@ declare namespace Handsontable {
+
+        /**
+         * Get hook bucket based on the context of the object or if argument is undefined, get the global hook bucket.
+         *
+         * @param context (optional) A Handsontable instance.
+         */
+        getBucket(context?: Handsontable): Object;
+
+        /**
+         * Returns an array of registered hooks.
+         */
+        getRegistered(): string[];
+
+        /**
+         * Checks whether there are any registered listeners for the provided hook name.
+         *
+         * @param key Hook name.
+         * @param context (optional) A Handsontable instance.
+         */
+        has(key: string, context?: Handsontable): boolean;
+
+        /**
+         * Returns a boolean depending on if a hook by such name has been registered.
+         *
+         * @param key Hook name.
+         */
+        isRegistered(key: string): boolean;
+
+        /**
+         * Adds a listener to a specified hook. After the hook runs this listener will be automatically removed from the bucket.
+         *
+         * @param key Hook/Event name.
+         * @param callback Callback function.
+         * @param context (optional) A Handsontable instance.
+         */
+        once(key: string, callback: Function | Function[], context?: Handsontable): void;
+
+        /**
+         * Registers a hook name (adds it to the list of the known hook names). Used by plugins.
+         *
+         * @param key The hook name.
+         */
         register(key: string): void;
-        run(instace: Core, hookName: string, key?: any, value?: any): any;
+
+        /**
+         * Removes a listener from a hook with a given name.
+         *
+         * @param key Hook/Event name.
+         * @param callback Callback function.
+         * @param context (optional) A Handsontable instance.
+         */
+        remove(key: string, callback: Function, context?: Handsontable): boolean;
+
+        /**
+         * Runs all local and global callbacks assigned to the hook identified by the key parameter.
+         * It returns either a return value from the last called callback or the first parameter (p1) passed to the run function.
+         *
+         * @param context Handsontable instance.
+         * @param key Hook/Event name.
+         * @param p1 (optional) Parameter to be passed as an argument to the callback function.
+         * @param p2 (optional) Parameter to be passed as an argument to the callback function.
+         * @param p3 (optional) Parameter to be passed as an argument to the callback function.
+         * @param p4 (optional) Parameter to be passed as an argument to the callback function.
+         * @param p5 (optional) Parameter to be passed as an argument to the callback function.
+         * @param p6 (optional) Parameter to be passed as an argument to the callback function.
+         */
+        run(context: Handsontable, key: string, p1?: any, p2?: any, p3?: any, p4?: any, p5?: any, p6?: any): any;
```

Comment:
That's a possibility, but based on [the documentation for the run method](https://docs.handsontable.com/0.31.2/Hooks.html#run), it seems that you are actually limited to just six parameters, so this seems to more accurately reflect this.